jerry-core/vm/vm.c

```diff
@@ -1721,16 +1721,20 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
           result = vm_op_get_value (left_value,
                                     right_value);
 
-          if (opcode < CBC_PRE_INCR)
+          if (ECMA_IS_VALUE_ERROR (result))
           {
-            left_value = ECMA_VALUE_UNDEFINED;
-            right_value = ECMA_VALUE_UNDEFINED;
-
-            if (ECMA_IS_VALUE_ERROR (result))
+            if (opcode < CBC_PRE_INCR)
             {
-              goto error;
+              left_value = ECMA_VALUE_UNDEFINED;
+              right_value = ECMA_VALUE_UNDEFINED;
             }
+            goto error;
+          }
 
+          if (opcode < CBC_PRE_INCR)
+          {
+            left_value = ECMA_VALUE_UNDEFINED;
+            right_value = ECMA_VALUE_UNDEFINED;
```

Comment:
Is the proposed approach best way to write this?

```suggestion
if (opcode < CBC_PRE_INCR)
{
  left_value = ECMA_VALUE_UNDEFINED;
  right_value = ECMA_VALUE_UNDEFINED;
}

if (ECMA_IS_VALUE_ERROR (result))
{
  goto error;
}
	
if (opcode < CBC_PRE_INCR)
{
  break;
}
```
Wouldn't this snippet involve less copy-paste?

Reply:
Good idea! I've applied it.